src/queue.rs

```diff
@@ -548,6 +563,78 @@ impl<M: GuestAddressSpace> Queue<M> {
         snapshot
             .write_obj(self.next_used.0 as u16, used_ring.unchecked_add(2))
             .unwrap();
+
+        Some(self.next_used.0)
+    }
+
+    /// Update avail_event on the used ring with the last index in the avail ring.
+    pub fn update_avail_event(&mut self) {
```

Comment:
I think this method should be split into an `avail_index()` that returns the available ring index, and then a `set_avail_event()` that writes a specified value to the `avail_event` location in the used ring.